akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -228,9 +228,10 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
           "value, because important history can be lost when replacing the `ORSet` and " +
           "undesired effects of merging will occur. Use `ORMultiMap` or `ORMap.updated` instead.")
     else {
-      val putDeltaOp = PutDeltaOp(keys.resetDelta.add(node, key).delta.get, key → value, zeroTag)
-      // put forcibly damages history, so we propagate full value that will overwrite previous values
-      new ORMap(keys.add(node, key), values.updated(key, value), zeroTag, Some(newDelta(putDeltaOp)))
+      val newKeys = keys.resetDelta.add(node, key)
```

Comment:
This is why the fix works. `keys.add(node, key)` has a nasty side-effect of increasing the `vvector` at a given node. So it has to be executed exactly once to avoid having two entries for the same key with different versions.